pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentReplicator.java

```diff
@@ -771,10 +784,55 @@ public boolean isConnected() {
         return producer != null && producer.isConnected();
     }
 
+    @Override
+    public CompletableFuture<Void> resetCursor(Position position) {
+        CURSOR_RESETTING_UPDATER.set(this, TRUE);
```

Comment:
Is there a race condition here when multiple threads call this method concurrently?

Reply:
the ``cursor.asyncResetCursor`` will ensure that.

Reply:
But there may be multiple threads calling `cursor.asyncResetCursor` concurrently as well. I think we need to use `compareAndSet` here. Please correct me if I have missed something.

Reply:
Well, looks like this state is useless here, the cursor will ensure the thread-safe.
So, I chose to delete it, What do you think about it?

Reply:
It's not deleted. Did you update your codes?